packages/components/src/unit-control/utils.js

```diff
@@ -151,7 +151,8 @@ export const DEFAULT_UNIT = allUnits.px;
  * @return {Array<number, string>} The extracted number and unit.
  */
 export function getParsedValue( value, unit, units ) {
-	const initialValue = unit ? `${ value }${ unit }` : value;
+	const initialValue =
+		unit && Number.isFinite( value ) ? `${ value }${ unit }` : value;
```

Comment:
What is the reasoning behind adding the `Number.isFinite( value )` check here?

Reply:
This was a remnant of an earlier slightly different attempt to get this working, so I've reverted the change, thanks for spotting it!